libs/ledger/src/chain/block_coordinator.cpp

```diff
@@ -1106,20 +1112,25 @@ BlockCoordinator::State BlockCoordinator::OnReset()
 
   if (next_block_)
   {
+    std::cerr << "wasn " << next_block_->body.block_number << std::endl;  // DELETEME_NH
     block = next_block_.get();
   }
   else if (current_block_)
   {
+    std::cerr << "wasc" << std::endl;  // DELETEME_NH
     block = current_block_.get();
   }
   else
   {
     FETCH_LOG_ERROR(LOGGING_NAME, "Unable to find a previously executed block!");
   }
 
+  // TODO(HUT): figure out where/when to do this
+  // block->UpdateDigest();
```

Comment:
Already done after collecting final merkle tree hash?

Reply:
There is a corner case when exceptions are thrown, this might not be true. Will open ticket.